### ISSUE TITLE
Unable to get OpenStack provider working

### ISSUE THREAD

**Issue by toddbaumann** [codedellemc/rexray#515]
I am trying to test out rexray for docker volumes in an OpenStack mitaka release.  

I deployed a regular Ubuntu 16.04 VM with docker.io installed and Rexray version 0.3.3 (the only one specifically supported).  Whenever I try and run it I get the following.  

```
ERRO[0000] default module(s) failed to start
INFO[0000] service sent registered modules start signals
ERRO[0000] service initialized failed
INFO[0000] error starting modul                          address=unix:///run/docker/plugins/rexray.sock inner=reflect: call of reflect.Value.Elem on struct Value name=default-docker typeName=docker
```

The contents of my config.yml file are 

```
rexray:
  StorageDrivers:
  - OpenStack
openstack:
  authUrl: https://OPENSTACKHOST:5000/v2.0
  username: admin
  password: #######
  regionName: RegionOne
  availabilityZoneName:  
```

The OpenStack has the cinder service available and is using KVM as the hypervisor (I saw references to xen being something of interest in other issues).  

```
rexray volume -l debug 

PANI[0000] reflect: call of reflect.Value.Elem on struct Value
panic: reflect: call of reflect.Value.Elem on struct Value [recovered]
    panic: reflect: call of reflect.Value.Elem on struct Value [recovered]
    panic: (*logrus.Entry) (0x1ad60a0,0xc8203b8740)

goroutine 1 [running]:
github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus.Entry.log(0xc82000a1c0, 0xc8203ea5a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc8203b8700, ...)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus/entry.go:113 +0x618
github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus.(*Entry).Panic(0xc8200a0678, 0xc8200a0760, 0x1, 0x1)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus/entry.go:169 +0xdf
github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus.(*Logger).Panic(0xc82000a1c0, 0xc8200a0760, 0x1, 0x1)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus/logger.go:163 +0xe7
github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus.Panic(0xc8200a0760, 0x1, 0x1)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus/exported.go:107 +0x41
github.com/emccode/rexray/rexray/cli.(*CLI).Execute.func1()
    /home/travis/gopath/src/github.com/emccode/rexray/rexray/cli/cli.go:224 +0x2e0
github.com/emccode/rexray/rexray/cli.(*CLI).execute.func1()
    /home/travis/gopath/src/github.com/emccode/rexray/rexray/cli/cli.go:243 +0xd4
reflect.Value.Elem(0x19655e0, 0xc820404760, 0x59, 0x0, 0x0, 0x0)
    /usr/local/go/src/reflect/value.go:732 +0x21f
github.com/emccode/rexray/vendor/github.com/akutz/golf.fore(0xc820425010, 0xf, 0x19655e0, 0xc820404760, 0xc8203ea510)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/akutz/golf/golf.go:170 +0x201
github.com/emccode/rexray/vendor/github.com/akutz/golf.fore(0xc820424e80, 0xb, 0x169f260, 0xc8203ea450, 0xc8203ea510)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/akutz/golf/golf.go:215 +0x84a
github.com/emccode/rexray/vendor/github.com/akutz/golf.fore(0x1b30d40, 0x5, 0x1a70cc0, 0xc820404780, 0xc8203ea510)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/akutz/golf/golf.go:215 +0x84a
github.com/emccode/rexray/vendor/github.com/akutz/golf.Fore(0x1b30d40, 0x5, 0x1a70cc0, 0xc820404780, 0xc82025af01)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/akutz/golf/golf.go:146 +0xad
github.com/emccode/rexray/vendor/github.com/akutz/golf/logrus.(*TextFormatter).printColored(0xc8201dcc80, 0xc8202d0380, 0xc8203b8640, 0xc8204047e0, 0x2, 0x2, 0x1d077c0, 0x19)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/akutz/golf/logrus/text_formatter.go:132 +0x509
github.com/emccode/rexray/vendor/github.com/akutz/golf/logrus.(*TextFormatter).Format(0xc8201dcc80, 0xc8203b8640, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/akutz/golf/logrus/text_formatter.go:79 +0x605
github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus.(*Entry).Reader(0xc8203b8640, 0xc8078c8805, 0x0, 0x0)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus/entry.go:44 +0x67
github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus.Entry.log(0xc82000a1c0, 0xc8203ea4e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x40c205, 0xc8204047c0, ...)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus/entry.go:94 +0x282
github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus.(*Entry).Debug(0xc8203b8600, 0xc8200a1a10, 0x1, 0x1)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/Sirupsen/logrus/entry.go:120 +0xcb
github.com/emccode/rexray/core.(*RexRay).InitDrivers(0xc820223680, 0x0, 0x0)
    /home/travis/gopath/src/github.com/emccode/rexray/core/rexray.go:89 +0x1990
github.com/emccode/rexray/rexray/cli.(*CLI).preRun(0xc8201ee600, 0xc82009ddc0, 0xc82023c1e0, 0x0, 0x2)
    /home/travis/gopath/src/github.com/emccode/rexray/rexray/cli/cli.go:328 +0x381
github.com/emccode/rexray/rexray/cli.(*CLI).(github.com/emccode/rexray/rexray/cli.preRun)-fm(0xc82009ddc0, 0xc82023c1e0, 0x0, 0x2)
    /home/travis/gopath/src/github.com/emccode/rexray/rexray/cli/cli.go:179 +0x48
github.com/emccode/rexray/vendor/github.com/spf13/cobra.(*Command).execute(0xc82009ddc0, 0xc82023c080, 0x2, 0x2, 0x0, 0x0)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/spf13/cobra/command.go:517 +0x911
github.com/emccode/rexray/vendor/github.com/spf13/cobra.(*Command).Execute(0xc82009c700, 0x0, 0x0)
    /home/travis/gopath/src/github.com/emccode/rexray/vendor/github.com/spf13/cobra/command.go:617 +0x46a
github.com/emccode/rexray/rexray/cli.(*CLI).execute(0xc8201ee600)
    /home/travis/gopath/src/github.com/emccode/rexray/rexray/cli/cli.go:247 +0x42
github.com/emccode/rexray/rexray/cli.(*CLI).Execute(0xc8201ee600)
    /home/travis/gopath/src/github.com/emccode/rexray/rexray/cli/cli.go:230 +0x3e
github.com/emccode/rexray/rexray/cli.Execute()
    /home/travis/gopath/src/github.com/emccode/rexray/rexray/cli/cli.go:204 +0x25
main.main()
    /home/travis/gopath/src/github.com/emccode/rexray/rexray/rexray.go:11 +0x14

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
    /usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 6 [syscall]:
os/signal.loop()
    /usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
    /usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 19 [IO wait]:
net.runtime_pollWait(0x7f34db5e6a00, 0x72, 0xc82000e180)
    /usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8202d00d0, 0x72, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8202d00d0, 0x0, 0x0)
    /usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8202d0070, 0xc8202e8000, 0x1000, 0x1000, 0x0, 0x7f34db5db050, 0xc82000e180)
    /usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8202c4010, 0xc8202e8000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/net.go:172 +0xe4
net/http.noteEOFReader.Read(0x7f34d8754000, 0xc8202c4010, 0xc8202e6058, 0xc8202e8000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8202c00e0, 0xc8202e8000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc8202d42a0)
    /usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc8202d42a0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8202e6000)
    /usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
    /usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 20 [select]:
net/http.(*persistConn).writeLoop(0xc8202e6000)
    /usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
    /usr/local/go/src/net/http/transport.go:686 +0xc9d
root@gh1404-inst1-ytktx6lsttc4:/etc/rexray#
```

**Comment by kacole2** [comment]
What is your version of Docker?

**Comment by toddbaumann** [comment]
On the Ubuntu 16.04 host 
```root@gh3-inst1-t52t5asbx4nm:~# docker --version
Docker version 1.12.0, build 8eab29e```
On the Ubuntu 14.04 host 
```root@gh1404-inst1-ytktx6lsttc4:~# docker --version
Docker version 1.6.2, build 7c8fca2```

Both fail in the same way (I spun up the 14.04 instance to make sure it wasn't something kernel)

**Comment by kacole2** [comment]
You're right that kernel shouldn't matter. You need to be at Docker 1.11+ so the 14.04 needs to be upgraded to work properly. 

I'm not an OpenStack expert by any means so @cduchesne @codenrhoden @justinclayton any ideas?

I saw in Slack there may be camelCase errors in the config

**Comment by toddbaumann** [comment]
I corrected the camelCase issues ( I had seen it done a few ways so tried it that way with no luck).   I can upgrade docker on the Ubuntu 14.04 and try it out.

**Comment by toddbaumann** [comment]
Tested with docker 1.12 on 1404 with same result.

**Comment by toddbaumann** [comment]
I was able to figure out why the service wasn't started.  It looks like there is no way to start the rexray service if your OpenStack instance has a self signed certificate.  I tested this with the insecure flag set to true in the OpenStack properties and still there was no way to get the service started.  

Based on that it looks like the insecure option was never merged in or something not sure if this is a bug or not but it looks like it is.  

This was tested on rexray version 0.3.0, 0.3.1 and 0.3.3 all with the same behavior
